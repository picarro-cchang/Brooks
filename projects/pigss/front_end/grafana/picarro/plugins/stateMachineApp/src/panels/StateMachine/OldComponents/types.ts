// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
