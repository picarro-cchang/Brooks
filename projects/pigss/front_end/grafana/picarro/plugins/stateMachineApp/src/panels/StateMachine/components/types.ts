export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
