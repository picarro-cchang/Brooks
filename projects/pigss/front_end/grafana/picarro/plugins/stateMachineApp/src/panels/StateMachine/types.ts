import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
