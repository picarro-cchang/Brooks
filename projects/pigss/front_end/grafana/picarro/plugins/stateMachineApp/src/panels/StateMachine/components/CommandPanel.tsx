import React, { Component, PureComponent } from "react";
import { CommandPanelOptions } from "./../types";

class CommandPanel extends PureComponent<CommandPanelOptions> {
  classNameOpt = {
    DISABLED: "",
    READY: "btn-outline-success",
    ACTIVE: "btn-green"
  };

  getDisabled = (element: string): boolean => {
    let disabled = true;
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      disabled = status_dict[element] === "DISABLED";
    }
    return disabled;
  };

  getClassNameOpt = (element: string): string => {
    let classNames = "";
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      classNames = (this.classNameOpt as any)[status_dict[element]];
    }
    return classNames;
  };

  render() {
    return (
      <div className="panel-command">
        <div style={{ width: "100%", marginTop: 20 }}>
          <div
            className="grid-command"
            style={{
              display: "grid",
              gridTemplateColumns: "1fr 1fr 1fr",
              gridGap: 5,
              padding: 20
            }}
          >
            <button
              onClick={e => this.props.ws_sender({ element: "standby" })}
              disabled={this.getDisabled("standby")}
              value="standby"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("standby")
              }
            >
              Standby
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "identify" })}
              disabled={this.getDisabled("identify")}
              value="identify"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("identify")
              }
            >
              Identify Available Channels
            </button>

            <button
              onClick={e => {
                this.props.ws_sender({ element: "plan" });
              }}
              disabled={this.getDisabled("plan")}
              value="plan"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("plan")
              }
            >
              Edit Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "run" })}
              disabled={this.getDisabled("run")}
              value="run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("run")
              }
            >
              Run Channel
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_run" })}
              disabled={this.getDisabled("plan_run")}
              value="plan_run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_run")
              }
            >
              Run Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_loop" })}
              disabled={this.getDisabled("plan_loop")}
              value="plan_loop"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_loop")
              }
            >
              Loop Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "reference" })}
              disabled={this.getDisabled("reference")}
              value="reference"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("reference")
              }
            >
              Reference
            </button>
            <button
              value="edit"
              className={"btn btn-large btn-edit btn-danger"}
              onClick={e => this.props.ws_sender({ element: "edit" })}
              disabled={this.getDisabled("edit")}
            >
              Edit Labels
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CommandPanel;
