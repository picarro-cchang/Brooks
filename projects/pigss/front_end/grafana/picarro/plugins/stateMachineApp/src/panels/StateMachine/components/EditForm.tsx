import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div className="row">
        <label className="edit-label"> Bank {num}</label>
        {/* <input
          name={"bank" + num}
          className="col-sm-6 edit-input"
          type="text"
          value={this.props.plan.bank_names[num].name}
          onChange={event =>
            this.props.handleBankChange(
              event.target.value,
              num.toString()
            )
          }
          maxLength={14}
        /> */}
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={8}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
