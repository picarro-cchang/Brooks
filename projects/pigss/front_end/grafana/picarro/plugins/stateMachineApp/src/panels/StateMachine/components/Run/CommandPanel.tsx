import React, { Component, PureComponent } from "react";
import { CommandPanelOptions } from "./../types";

class CommandPanel extends PureComponent<CommandPanelOptions, any> {
  classNameOpt = {
    DISABLED: "",
    READY: "btn-outline-success",
    ACTIVE: "btn-green"
  };

  getDisabled = (element: string): boolean => {
    let disabled = true;
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      disabled = status_dict[element] === "DISABLED";
    }
    return disabled;
  };

  getClassNameOpt = (element: string): string => {
    let classNames = "";
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      classNames = (this.classNameOpt as any)[status_dict[element]];
    }
    return classNames;
  };

  render() {
    console.log("Hello from Command ", this.props.plan)
    return (
      <div className="panel-command">
        <div style={{ width: "100%", marginTop: 20 }}>
          <div
            className="grid-command"
            style={{
              display: "grid",
              gridTemplateColumns: "4rem 4rem 4rem 4rem 4rem 4rem",
              gridGap: 5,
              padding: 20
            }}
          >
            <button
              id={"standby"}
              onClick={e => {
                this.props.ws_sender({ element: "standby" });
              }}
              disabled={this.getDisabled("standby")}
              value="standby"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("standby")
              }
            >
              Standby
            </button>
            <button
              id={"identify"}
              onClick={e => this.props.ws_sender({ element: "identify" })}
              disabled={this.getDisabled("identify")}
              value="identify"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("identify")
              }
            >
              Identify Available Channels
            </button>

            <button
              id={"edit-plan"}
              onClick={e => {
                this.props.layoutSwitch();
              }}
              disabled={this.getDisabled("plan")}
              value="plan"
              className={
                "btn btn-large btn-command btn-edit-plan " +
                this.getClassNameOpt("plan")
              }
            >
              Edit Plan
            </button>
            <button
              id={"load-plan"}
              onClick={e => {
                this.props.ws_sender({ element: "load" })
                this.props.updatePanel(1)
              }}
              disabled={this.getDisabled("plan")}
              value="load"
              className={
                "btn btn-large btn-command btn-load " +
                this.getClassNameOpt("plan")
              }
            >
              Load Plan
            </button>
            <button
              id={"run-channel"}
              onClick={e => this.props.ws_sender({ element: "run" })}
              disabled={this.getDisabled("run")}
              value="run"
              className={
                "btn btn-large btn-command btn-run-single " +
                this.getClassNameOpt("run")
              }
            >
              Single Port
            </button>
            <button
              id={"run-plan"}
              onClick={e => {
                this.props.ws_sender({ element: "plan_run" })
                this.props.setModalInfo(true, `<div>Run Plan ${this.props.plan.plan_filename} starting at Step ${this.props.plan.current_step}?</div>`, 3, {
                  1: {
                    caption: "Ok",
                    className: "btn btn-success btn-large",
                    response: {plan: this.props.plan}
                  },
                  2: {
                    caption: "Start at Step 1",
                    className: "btn btn-success btn-large",
                    response: {step: 1, plan: this.props.plan}
                  },
                  3: {
                    caption: "Cancel",
                    className: "btn btn-danger btn-large",
                    response: null
                  }
                }, 'loopPlan')
              }}
              disabled={this.getDisabled("plan_run")}
              value="plan_run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_run")
              }
            >
              Run Plan
            </button>
            <button
              id={"loop-plan"}
              onClick={e => {
                console.log("Setting loop plan ", this.state)
                this.props.setModalInfo(true, `<div>Loop Plan ${this.props.plan.plan_filename} starting at Step ${this.props.plan.current_step}?</div>`, 3, {
                  1: {
                    caption: "Ok",
                    className: "btn btn-success btn-large",
                    response: {plan: this.props.plan}
                  },
                  2: {
                    caption: "Start at Step 1",
                    className: "btn btn-success btn-large",
                    response: {step: 1, plan: this.props.plan}
                  },
                  3: {
                    caption: "Cancel",
                    className: "btn btn-danger btn-large",
                    response: null
                  }
                }, 'runPlan')
                this.props.ws_sender({ element: "plan_loop" })
              }}
              disabled={this.getDisabled("plan_loop")}
              value="plan_loop"
              className={
                "btn btn-large btn-command btn-run-loop " +
                this.getClassNameOpt("plan_loop")
              }
            >
              Loop Plan
            </button>
            <button
              id={"reference"}
              onClick={e => this.props.ws_sender({ element: "reference" })}
              disabled={this.getDisabled("reference")}
              value="reference"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("reference")
              }
            >
              Reference
            </button>
            <button
              id={"edit-labels"}
              value="edit"
              className={"btn btn-large btn-edit btn-danger"}
              onClick={e => this.props.updatePanel(2)}
              disabled={this.getDisabled("edit")}
            >
              Edit Labels
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CommandPanel;
