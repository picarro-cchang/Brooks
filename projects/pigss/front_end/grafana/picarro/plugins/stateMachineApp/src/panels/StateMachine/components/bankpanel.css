.bank-btn {
    border-radius: 25%;
    /*border: 4px solid #bbbbbbbd;*/
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    height: 90px;
    width: 90px;
    color: black;
    position: relative;
}
.chn-label {
    position: absolute;
    top: 20px;
    left: -0.37em;
    bottom: 0;
    right: 0;
}
.btn-clean {
    width:100%;
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    color: black;
    border-radius: 10px;
}
.btn-clean-act {
    border: none;
}

.test {
    color: black;
}
.btn-group {
    width: 100%;
    /*height: 35px;*/
    border-radius: 10%;
    text-align: center;
    /*margin-right: 20px;*/
    margin-left: -20px;
    font-size: 17px;
}

.btn-row-1 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-2 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-3 {
    margin-top: 10px;
    align-items: center;
    style: flex;
    justify-content: center;
}
.panel-plan {
    height: 640px;
    /*background-color: #888;*/
    /*border: 3px solid #111;*/
    /*border-radius: 7px;*/
    /*background-color: #82828287;*/
    background-color: #cacaca;
    padding: 10px;
    /*border: 2px solid #1111117a;*/
    border-radius: 5px;
}
.panel-plan-inner {
    overflow-x: hidden;
    overflow-y: auto;
    max-height: 420px;
    border-radius: 3px;
}
.plan-input {
    background-color: white;
    border: 1px solid black;
    border-radius: 2px;
    /*margin-right: -20px;*/
    color: black;
    margin-bottom: 5px;
}
.panel-bank {
    /*border: 2px solid #111;*/
    border: 2px solid #1111117a;
    margin-right: 2%;
    border-radius: 10px;
    padding:10px;
    float: left;
    background-color: rgba(136, 136, 136, 0.51);

}
.grid-bank {
    display: grid;
    grid-gap: 10px;
    grid-template-columns: 1fr 1fr;
    padding-bottom: 10px;
}
.btn-green {
    background-color:  rgb(86, 166, 75);
}
.btn-1-3 {
    grid-column-start: 1;
    grid-column-end: 4;
    width: 100%;
}
.reference-btn {
    display: inline-block;
}
.ref-btn {
    width: 400px;
    grid-row-start: 2;
    grid-column-start: 1;
    /*margin-top: -100px;*/
}

.btn-command {
    border-radius: 5px;
    color: black;
    height: 80px;
}
.btn-edit {
    border-radius: 5px;
    color: white;
    height: 55px;
    width: 100px;
    margin-right: 20px;
    background-color: red;
    background-image: none;
}
.btn-cancel {
    background-image: none;
    background-color: red;
}
.edit-panel {
    color: black;
    background: white;
    border: 1px solid black;
    border-radius: 2px;
}
.edit-label {
    margin-left: 30px;
    margin-right: 10px;
    font-size: 15px;
    color: black;
}
.panel-edit {
    height: 550px;
    background-color: #cacaca;
    padding: 10px;
    border: 3px solid #111;
    border-radius: 7px;
}
.btn-edit-panel {
    margin-bottom: 10px;
    width: 95px;
    height: 40px;
    border-radius: 5px;
    text-align: center;
    font-size: 20px;
}
.btn-run {
    width: 100%;
}
.btn-inverse.disabled {
    background-color: #333;
    opacity: .9;
}
.btn-success {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;
}

.btn-success:active, .btn-success:focus {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;

}
.btn-run-active {
    background-color: rgba(98, 100, 104, 0.86);
}
.btn-run-issue {
    background-color: #716e6e;
}
.btn-chann {
     background-color:  rgb(86, 166, 75);
 }
.col-bank-name {
    /*max-width: 45%;*/
    /*min-width: 20%;*/
    text-align:left;
    padding-left: 5px;
    padding-right: 5px;
}
.input-small {
    width: 70px;
    margin-left: -10px;
}

.radio-btn {
    display: block;
    position: relative;
    /*padding-left: 35px;*/
    margin-bottom: 12px;
    /*margin-left: 20px;*/
    cursor: pointer;
    font-size: 22px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}
.radio-btn input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}

.checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border-radius: 50%;
}

.radio-btn:hover input ~ .checkmark {
    background-color: #565657;
}

.radio-btn input:checked ~ .checkmark {
    background-color: rgb(31, 96, 196);
}


.checkmark:after {
    content: "";
    position: absolute;
    display: none;
}

.radio-btn input:checked ~ .checkmark:after {
    display: block;
}

.radio-btn .checkmark:after {
    top: 9px;
    left: 9px;
    width: 8px;
    height: 8px;
    border-radius: 50%;
    background: white;
}
.button-edit {
    justify-content: center;
}

.cancel:before {
    content: '\2716';
}

.cancel {
    position: absolute;
    top: 0;
    right: 10%;
    cursor: pointer;
    color: black;
    font-size: 25px;
    display: inline-block;
}

