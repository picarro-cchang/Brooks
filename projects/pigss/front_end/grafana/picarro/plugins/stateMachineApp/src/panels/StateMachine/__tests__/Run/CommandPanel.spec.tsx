import React from "react";
import CommandPanel from "./../../components/Run/CommandPanel";
import { CommandPanelOptions } from "./../../components/types";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import WS from "jest-websocket-mock";
import mockPlan from "./../../api/__mocks__/mockPlan.json";

const mockClick = jest.fn(element => {
  return element;
});
const mockUpdatePanel = jest.fn(e => {return e});
const mockSwitchLayout = jest.fn();
const mockSetModalInfo = jest.fn();
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;

const defaultProps: CommandPanelOptions = {
  planID: "",
  uistatus: {
    standby: "ACTIVE ",
    identify: "DISABLED",
    run: "READY",
    plan: "READY",
    plan_run: "READY",
    plan_loop: "READY",
    reference: "READY",
    edit: "READY"
  },
  ws_sender: mockClick,
  plan: mockPlan,
  updatePanel: mockUpdatePanel,
  layoutSwitch: mockSwitchLayout,
  loadedFileName: "Testing"
};

describe("<CommandPanel />", () => {
  const wrapper = shallow(<CommandPanel {...defaultProps} />);
  const instance = wrapper.instance() as CommandPanel;
  const server = new WS(socketURL);
  const client = new WebSocket(socketURL);

  it("Renders Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Buttons Enabled/Disabled Accordingly", () => {
    const value = wrapper.find("button#identify").props().disabled;
    expect(value).toEqual(true);
  });

  it("getDisabled functionality", () => {
    expect(instance.getDisabled("standby")).toEqual(false);
    expect(instance.getDisabled("status")).toEqual(true);
  });

  it("getClassNameOpt functionality", () => {
    expect(instance.getClassNameOpt("status")).toEqual("");
  });

  it("Standby", async () => {
    await server.connected;

    wrapper.find("button#standby").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("standby");
    expect(mockClick).toHaveReturnedWith({ element: "standby" });
    expect(server).toHaveReceivedMessages(["standby"]);
    mockClick.mockClear();
  });

  it("Identify", async () => {
    await server.connected;

    wrapper.find("button#identify").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("identify");
    expect(mockClick).toHaveReturnedWith({ element: "identify" });
    expect(server).toHaveReceivedMessages(["identify"]);
    mockClick.mockClear();
  });

  it("Edit Plan", async () => {
    wrapper.find("button#edit-plan").simulate("click");
    expect(mockSwitchLayout).toHaveBeenCalled();
    mockSwitchLayout.mockClear();
  });

  it("Run Channel", async () => {
    await server.connected;

    wrapper.find("button#run-channel").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("run");
    expect(mockClick).toHaveReturnedWith({ element: "run" });
    expect(server).toHaveReceivedMessages(["run"]);
    mockClick.mockClear();
  });

  it("Run Plan", async () => {
    await server.connected;

    wrapper.find("button#run-plan").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("plan_run");
    expect(mockClick).toHaveReturnedWith({ element: "plan_run" });
    expect(server).toHaveReceivedMessages(["plan_run"]);
    mockClick.mockClear();
  });

  it("Loop Plan", async () => {
    await server.connected;

    wrapper.find("button#loop-plan").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("plan_loop");
    expect(mockClick).toHaveReturnedWith({ element: "plan_loop" });
    expect(server).toHaveReceivedMessages(["plan_loop"]);
    mockClick.mockClear();
  });

  it("Reference", async () => {
    await server.connected;

    wrapper.find("button#reference").simulate("click");
    const element = mockClick.mock.calls[0][0].element;

    client.send(element);
    expect(mockClick).toHaveBeenCalled();
    await expect(server).toReceiveMessage("reference");
    expect(mockClick).toHaveReturnedWith({ element: "reference" });
    expect(server).toHaveReceivedMessages(["reference"]);
    mockClick.mockClear();
  });

  it("Edit Labels", async () => {
    wrapper.find("button#edit-labels").simulate("click");
    expect(mockUpdatePanel).toHaveBeenCalledWith(2);
    mockUpdatePanel.mockClear();
  });
});
