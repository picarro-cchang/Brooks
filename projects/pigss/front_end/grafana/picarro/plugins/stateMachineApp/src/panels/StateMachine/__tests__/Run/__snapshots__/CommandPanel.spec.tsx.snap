// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommandPanel /> Renders Correctly 1`] = `
<div
  className="panel-command"
>
  <div
    style={
      Object {
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <div
      className="grid-command"
      style={
        Object {
          "display": "grid",
          "gridGap": 5,
          "gridTemplateColumns": "4rem 4rem 4rem 4rem 4rem 4rem",
          "padding": 20,
        }
      }
    >
      <button
        className="btn btn-large btn-command btn-1-3 undefined"
        disabled={false}
        id="standby"
        onClick={[Function]}
        value="standby"
      >
        Standby
      </button>
      <button
        className="btn btn-large btn-command btn-1-3 "
        disabled={true}
        id="identify"
        onClick={[Function]}
        value="identify"
      >
        Identify Available Channels
      </button>
      <button
        className="btn btn-large btn-command btn-edit-plan btn-outline-success"
        disabled={false}
        id="edit-plan"
        onClick={[Function]}
        value="plan"
      >
        Edit Plan
      </button>
      <button
        className="btn btn-large btn-command btn-load btn-outline-success"
        disabled={false}
        id="load-plan"
        onClick={[Function]}
        value="load"
      >
        Load Plan
      </button>
      <button
        className="btn btn-large btn-command btn-run-single btn-outline-success"
        disabled={false}
        id="run-channel"
        onClick={[Function]}
        value="run"
      >
        Single Port
      </button>
      <button
        className="btn btn-large btn-command btn-run btn-outline-success"
        disabled={false}
        id="run-plan"
        onClick={[Function]}
        value="plan_run"
      >
        Run Plan
      </button>
      <button
        className="btn btn-large btn-command btn-run-loop btn-outline-success"
        disabled={false}
        id="loop-plan"
        onClick={[Function]}
        value="plan_loop"
      >
        Loop Plan
      </button>
      <button
        className="btn btn-large btn-command btn-1-3 btn-outline-success"
        disabled={false}
        id="reference"
        onClick={[Function]}
        value="reference"
      >
        Reference
      </button>
      <button
        className="btn btn-large btn-edit btn-danger"
        disabled={false}
        id="edit-labels"
        onClick={[Function]}
        value="edit"
      >
        Edit Labels
      </button>
    </div>
  </div>
</div>
`;
